specs/agents/metadata.md

```diff
@@ -22,6 +22,18 @@ System metadata relates to the host/container in which the service being monitor
    - pod name
    - pod UID
 
+#### Hostname
+
+The `system.hostname` field is deprecated in favour of two newer fields:
+- `system.configured_hostname` - the configured name of the host the monitored service is running on. It should only be 
+sent when configured by the user through the `ELASTIC_APM_HOSTNAME` config. 
+If provided, it is used as the event's hostname.
+- `system.detected_hostname` - the hostname detected by the APM agent. Wherever applicable, this field should contains 
+what the `hostname` command returns on the host machine, as it is mapped by the APM Server to the [`host.hostname` ECS field](https://www.elastic.co/guide/en/ecs/current/ecs-host.html#field-host-hostname). 
```

Comment:
What's the command to use for Windows? Should agents use environment variables as fallbacks? If so, which ones and what's the precedence?

Reply:
I think it's good enough for now to only try the `hostname` command and if this does not succeed, try any fallback that makes most sense in each agent. 
The ECS definition for `host.hostname` reads:
```
It normally contains what the hostname command returns on the host machine.
```

Do you know if Metricbeat is using any fallback for Windows?

Reply:
on Windows there seems to be a `hostname.exe` [according to that discussion](https://stackoverflow.com/questions/7348711/recommended-way-to-get-hostname-in-java)

Reply:
> on Windows there seems to be a hostname.exe according to that discussion

We can use that. 
Looking into this same discussion, we can also rely on standard env variables (`COMPUTERNAME` on Windows and `HOSTNAME` on Linux). 
The Java agent already [does that](https://github.com/elastic/apm-agent-java/blob/f9267ff7fc7df619cb2272b8790fde618055c8f6/apm-agent-core/src/main/java/co/elastic/apm/agent/impl/payload/SystemInfo.java#L228-L238) (and looking for the `HOST` env variable in addition), but unfortunately only as a fallback when failing to get the hostname through DNS resolution.

Should I add this to the spec? (👍  or 👎  would be good)

Reply:
+1 on deciding on the same strategy across agents which fallbacks (if any) should be applied. I'm not sure which fallbacks to use, though. But we should probably replicate the same behavior that Metricbeat has.